### DISCUSSION TITLE
rbenv install 3.2.4 BUILD FAILED (macOS 14.4.1 on arm64 using ruby-build 20240423)

### DISCUSSION THREAD

**Discussion by HiramaniPatel** [rbenv/ruby-build#2373] [Build failures]
fail via download ruby Version 3.2.4

BUILD FAILED (macOS 14.4.1 on arm64 using ruby-build 20240423)
[ruby-build.20240429121256.28842.log](https://github.com/rbenv/ruby-build/files/15145918/ruby-build.20240429121256.28842.log)

**Comment by andreydro** [comment]
<s>Having the same when do `rbenv install 2.7.6` </s>

Updated xcode and Xcode Command Line Tools - it helped

**Comment by mislav** [comment]
The error I see in the provided log is this one:
```
1522   │ readline.c:1903:37: error: use of undeclared identifier 'username_completion_function'; did you mean 'rl_username_completion_function'?
1523   │                                     rl_username_completion_function);
1524   │                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1525   │                                     rl_username_completion_function
1526   │ readline.c:79:42: note: expanded from macro 'rl_username_completion_function'
1527   │ # define rl_username_completion_function username_completion_function
1528   │                                          ^
1529   │ /usr/local/opt/readline/include/readline/readline.h:494:14: note: 'rl_username_completion_function' declared here
1530   │ extern char *rl_username_completion_function (const char *, int);
1531   │              ^
1532   │ linking shared-object rbconfig/sizeof.bundle
1533   │ ld: warning: ignoring duplicate libraries: '-lruby.3.2'
1534   │ 1 error generated.
1535   │ make[2]: *** [readline.o] Error 1
1536   │ make[1]: *** [ext/readline/all] Error 2
```

This was reported many times in the past, but I could never find a conclusive reason why it happens. You could search the issue backlog for potential workarounds, e.g. https://github.com/rbenv/ruby-build/issues/1829#issuecomment-983123112

**Comment by oclef** [comment]
I had the same issue with the exact same error.  I was able to solve the problem by running Terminal is Rosetta mode

Right Click Terminal > Get Info > Check Open using Rosetta

**Reply by aimansaeedsh** [reply]
Worked for me!

**Reply by oclef** [reply]
Cool.  Make sure to uncheck Open using Rosetta after building
